src/components/CompareResults/loader.ts

```diff
@@ -249,6 +249,7 @@ type DeferredLoaderData = {
   newRepos: Repository['name'][];
   frameworkId: Framework['id'];
   frameworkName: Framework['name'];
+  intervalValue: TimeRange['value'];
```

Comment:
I really don't like that we're lying here: that we don't return it ever.

I understand why you're doing it but I think it may not be needed if you write the code like this:

```js
const loaderData = useLoaderData() as XXX
if (loaderData.view === overTimeView) {
  const { newRev, intervalValue, ... } = loaderData; // Here typescript should know it's correct 
}
if (loaderData.view === compareView) {
  const { baseRev, ...} = loaderData; // Same
}
```

Can you test this and see if this works?

If that doesn't work you can also add `as XXX` after `loaderData` to force the type, but I _think_ this isn't needed.

I just noticed you did something similar in the subtests file.

_otherwise_, if that's really needed, we can use the same shape in all return values (that is the same properties) but don't lie about their types. Here for example `intervalValue` is always `undefined` (that is `void` in typescript) so we should type it this way.

Reply:
This comment applies to the changes to the other loader as well.